### ISSUE TITLE
add piglins/hoglins can spawn in soul light

### ISSUE THREAD

**Issue by W1ll-Du** [gnembon/fabric-carpet#1261]
[not my idea](https://feedback.minecraft.net/hc/en-us/community/posts/360071893811-Hoglins-Piglins-Spawning-Conditions-Should-Be-Reworked)
was a bit annoyed about piglins/hoglins while mining for debris and came across this post while thinking about it
essentially either a toggle or like can spawn in x light level, split between both mobs.
e.g. piglinSoulLightLevel 0 (11 default)
hoglinSoulLightLevel 15 (default)
it seems to me just adding an extra check in the spawning algorithm, though i dont know how complex it would be in actuality

**Comment by Ghoulboy78** [comment]
tbh it sounds about as complex to me as it does to you, unless you add the specific requirement of soul light level, in which case the difficulty jumps considerably. TBH it doesn't sound like a particularly bad idea, especially given that in 1.18+ mobs need light level 0 to spawn.

**Comment by W1ll-Du** [comment]
Yeah, the problem would be they way handles lighting, because if I remember correctly, the data is precalculated/saved so you'd either have to save the soul light elsewhere or just search a diamond thingy with radius 15 (around 9000 blocks) which just doesn't seem efficient enough to me. What do I know, though

**Comment by Ghoulboy78** [comment]
nah you're right about the soul light thing, except it's worse cos of how light goes around corners. As such, I think the soul light thing may not be worth it, I would prefer just some boolean option like `piglingsSpawnInLight` and `hoglinsSpawnInLight` which both default to true, and when they are false then piglins and hoglins would spawn normally.
